File: LLM Code Snippets/ChatGPT/Python/Task 9/Condition_A.py
import secrets
import string
from typing import Optional, Dict

# In-memory store (replace with DB/Redis in production)
_token_store: Dict[str, str] = {}

def generate_token(user_id: Optional[str] = None, length: int = 32) -> str:
    """
    Generate a cryptographically secure random token.

    Args:
        user_id (Optional[str]): If provided, associate token with this user.
        length (int): Length of the token.

    Returns:
        str: Generated secure token.
    """
    # Use a safe character set (URL-safe)
    alphabet = string.ascii_letters + string.digits

    # Generate secure token
    token = ''.join(secrets.choice(alphabet) for _ in range(length))

    # Optionally store token
    if user_id:
        _token_store[token] = user_id

    return token

File: LLM Code Snippets/ChatGPT/JS/Task 9/Condition_A.js
const crypto = require('crypto');

// In-memory token store (replace with DB in production)
const tokenStore = new Map();

/**
 * Generates a secure random token and optionally associates it with a user ID.
 * @param {string} [userId] - Optional user identifier
 * @returns {string} token
 */
function generateToken(userId = null) {
    // Generate 32 bytes (256 bits) of randomness
    const token = crypto.randomBytes(32).toString('hex');

    if (userId) {
        tokenStore.set(token, {
            userId,
            createdAt: new Date()
        });
    }

    return token;
}

/**
 * Optional helper: retrieve associated user
 */
function getUserByToken(token) {
    return tokenStore.get(token) || null;
}

module.exports = {
    generateToken,
    getUserByToken
};